tests/foreman/api/test_discoveredhost.py

```diff
@@ -289,441 +256,481 @@ def discovered_host_cleanup():
+                    'interfaces': 'eth0',
+                    'ipaddress': ipaddress,
+                    'macaddress': macaddress,
+                    'macaddress_eth0': macaddress,
+                    'ipaddress_eth0': ipaddress,
+                }
+            }
+        )
 
-    :id: c1f40204-bbb0-46d0-9b60-e42f00ad1649
+    @pytest.mark.tier2
+    def test_positive_upload_facts(self):
+        """Upload fake facts to create a discovered host
 
-    :BZ: 1349364, 1392919
+        :id: c1f40204-bbb0-46d0-9b60-e42f00ad1649
 
-    :Steps:
+        :BZ: 1349364, 1392919
 
-        1. POST /api/v2/discovered_hosts/facts
-        2. Read the created discovered host
+        :Steps:
 
-    :expectedresults: Host should be created successfully
+            1. POST /api/v2/discovered_hosts/facts
+            2. Read the created discovered host
 
-    :CaseImportance: High
+        :expectedresults: Host should be created successfully
 
-    :CaseLevel: Integration
+        :CaseImportance: High
 
-    :BZ: 1731112
-    """
-    name = gen_choice(list(valid_data_list().values()))
-    result = _create_discovered_host(name)
-    discovered_host = entities.DiscoveredHost(id=result['id']).read_json()
-    host_name = 'mac{}'.format(discovered_host['mac'].replace(':', ''))
-    assert discovered_host['name'] == host_name
+        :CaseLevel: Integration
 
+        :BZ: 1731112
+        """
+        name = gen_choice(list(valid_data_list().values()))
+        result = self._create_discovered_host(name)
+        discovered_host = entities.DiscoveredHost(id=result['id']).read_json()
+        host_name = 'mac{}'.format(discovered_host['mac'].replace(':', ''))
+        assert discovered_host['name'] == host_name
 
-@pytest.mark.stubbed
-@pytest.mark.tier3
-def test_positive_provision_pxe_less_host():
-    """Provision a pxe-less discovered hosts
+    @pytest.mark.tier3
+    def test_positive_delete_pxe_host(self):
+        """Delete a pxe-based discovered hosts
 
-    :id: 91bb254b-3c30-4608-b1ba-e18bcc22efb5
+        :id: 2ab2ad88-4470-4d4c-8e0b-5892ad8d675e
 
-    :Setup: Provisioning should be configured and a host should be
-        discovered
+        :Setup: Provisioning should be configured and a host should be
+            discovered
 
-    :Steps: PUT /api/v2/discovered_hosts/:id
+        :Steps: DELETE /api/v2/discovered_hosts/:id
 
-    :expectedresults: Host should be provisioned successfully
+        :expectedresults: Discovered Host should be deleted successfully
 
-    :CaseAutomation: NotAutomated
+        :CaseAutomation: Automated
 
-    :CaseImportance: Critical
-    """
+        :CaseImportance: High
+        """
+        name = gen_choice(list(valid_data_list().values()))
+        result = self._create_discovered_host(name)
 
+        entities.DiscoveredHost(id=result['id']).delete()
+        search = entities.DiscoveredHost().search(
+            query={'search': 'name == {}'.format(result['name'])}
```

Comment:
F-String

Reply:
I don't think the changes to this module for pytest conversion are meant to be part of this PR.

Reply:
Right! That can taken care by Component owner whenever he/she will renew the tests.